akka-http-circe/src/main/scala/de/heikoseeberger/akkahttpcirce/CirceSupport.scala

```diff
@@ -17,50 +17,83 @@
 package de.heikoseeberger.akkahttpcirce
 
 import akka.http.scaladsl.marshalling.{ Marshaller, ToEntityMarshaller }
+import akka.http.scaladsl.model.HttpEntity
 import akka.http.scaladsl.model.MediaTypes.`application/json`
 import akka.http.scaladsl.unmarshalling.{ FromEntityUnmarshaller, Unmarshaller }
 import akka.util.ByteString
-import io.circe.{ jawn, Decoder, Encoder, Json, Printer }
+import io.circe.{ jawn, Decoder, Errors, Printer, RootEncoder }
 
 /**
   * Automatic to and from JSON marshalling/unmarshalling using an in-scope *Circe* protocol.
+  * The unmarshaller fails-fast, throwing the first `Error` encountered.
   *
-  * To use automatic codec derivation, user need to import `circe.generic.auto._`.
+  * To use automatic codec derivation, user need to import `io.circe.generic.auto._`.
   */
-object CirceSupport extends CirceSupport
+object CirceSupport extends CirceSupport with NoSpacesPrinter with FailFastUnmarshaller
 
 /**
-  * JSON marshalling/unmarshalling using an in-scope *Circe* protocol.
+  * Automatic to and from JSON marshalling/unmarshalling using an in-scope *Circe* protocol.
+  * The unmarshaller accumulates all errors in the exception `Errors`.
   *
-  * To use automatic codec derivation, user need to import `io.circe.generic.auto._`
+  * To use automatic codec derivation, user need to import `io.circe.generic.auto._`.
   */
+object ErrorAccumulatingCirceSupport
+    extends CirceSupport
+    with NoSpacesPrinter
+    with ErrorAccumulatingUnmarshaller
+
 trait CirceSupport {
 
-  private val jsonStringUnmarshaller =
+  /**
+    * Printer used in the JSON marshaller.
+    */
+  def printer: Printer
+
+  private[akkahttpcirce] val stringUnmarshaller =
```

Comment:
It's just there for optimization and code re-use. 
I think it is quite common to _import_ the implicits by extending a trait, having them in scope in all the subclasses (including transitively). Sealing the trait would forbid this pattern.

The following classes hierarchy would be nice. Thoughts?
```scala
// Give both the trait and the object import patterns
trait FailFastCirceSupport extends BaseCirceSupport with NoSpacesPrinter with FailFastUnmarshaller
object FailFastCirceSupport extends FailFastCirceSupport

// Give both the trait and the object import patterns
trait ErrorAccumulatingCirceSupport extends BaseCirceSupport with NoSpacesPrinter with ErrorAccumulatingUnmarshaller
object ErrorAccumulatingCirceSupport extends ErrorAccumulatingCirceSupport

// Keep full backward compatibility 
@deprecated trait CirceSupport extends FailFastCirceSupport
@deprecated object CirceSupport extends CirceSupport

// Base trait with abstract members
trait BaseCirceSupport 

// same 
trait FailFastUnmarshaller { this: BaseCirceSupport => ??? }
trait ErrorAccumulatingUnmarshaller { this: BaseCirceSupport => ??? } 
trait NoSpacesPrinter { this: BaseCirceSupport => ??? }
```